### ISSUE TITLE
Use dependabot to auto monitor dependencies

### ISSUE THREAD

**Issue by cjyetman** [RMI-PACTA/practices#11]
I strongly encourage enabling Dependabot on repos that you maintain, if for nothing else to monitor GH actions that have been updated.

**Comment by jdhoffa** [comment]
I don't know much about how to use Dependabot/ what it can or can't do (beyond the action monitoring which I have seen before).

May I assign this issue to you, and can you elaborate on what other functionalities you think are useful from Dependabot?

**Comment by cjyetman** [comment]
This is the best starting point (for using Dependabot for version updates):
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/about-dependabot-version-updates

**Comment by cjyetman** [comment]
Dependabot can periodically check the versions of packages used in various languages in your repo and warn you when something has an update. In our repos, I've used it primarily to monitor GitHub actions and JavaScript/NPM modules. When it notices something has an update, like a version bump on an external GitHub action you are using, it automatically opens a PR in your repo with the appropriate change to target the new version.

@jdhoffa I'm not sure what it means to have this "assigned" to me. What is the task?

**Comment by jdhoffa** [comment]
Yup fair enough, mainly I just wanted more context here and thought you were the person to give it! Un-assigning you.

In any case, reading the additional context here, I believe this is too prescriptive to live as a `practice` (although agree that we may want to incentivize people to use this):
https://github.com/RMI-PACTA/practices/blob/fd6129122c8724c16faa0f0444ad9b981046f85d/README.md?plain=1#L5

In this repository, the goal is to cover the WHAT (e.g. ensure dependencies in your repositories are up to date) and WHY (e.g. to stay current with security issues, bug-fixes, new features, etc.)

without being prescriptive on the HOW (e.g. use dependabot). 

@AlexAxthelm do you think having a default `dependabot.yml` would be something that could appropriately live in the https://github.com/RMI-PACTA/Actions repo? I guess it isn't technically a GitHub action, although it feels "action adjacent". If you agree, I would transfer this issue to that repo. Otherwise I will tag it as `wontfix` and keep it open for conversation.

**Comment by AlexAxthelm** [comment]
Seems close enough that it should probably be included here.

**Comment by cjyetman** [comment]
does this repo allow non-prescriptive suggestions on the HOW? maybe that's the best in-between?

**Comment by AlexAxthelm** [comment]
Maybe something like

```
> **SUGGESTION:_**  The note content.
```

So we could do 

> **SUGGESTION:** The default R actions ([example](https://github.com/RMI-PACTA/actions/blob/main/examples/R.yml)) in [RMI-PACTA/actions](https://github.com/RMI-PACTA/actions) implements a check for this practice.

**Comment by jdhoffa** [comment]
Lol, well you both clearly didn't take the time to read the 2 sentences of the `README` that I linked XD

> we... recommend optional resources about _how_ to do it.


How this is done is clear, for example, in the `git` practices:

![Screenshot 2024-02-26 at 11 59 48](https://github.com/RMI-PACTA/practices/assets/8741309/22f42408-2768-4dd6-8d74-353d87bd31b1)

**Comment by jdhoffa** [comment]
So I think this could live in a documented practice regarding dependencies overall.

@cjyetman are you comfortable if I change the name of this issue to:
"Define `dependencies` practices"?

**Comment by cjyetman** [comment]
The README says...
> only recommend optional resources about _how_ to do it

so, I don't see how this does not fit here, but also don't really care
